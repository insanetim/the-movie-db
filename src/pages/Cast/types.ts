import { ErrorMsg } from 'src/interfaces/global.interface'
import { IMovieDetailsEx } from 'src/interfaces/movie.interface'

export type CastHookReturn = {
  error: ErrorMsg
  loading: boolean
  movie: IMovieDetailsEx | undefined
  movieSlug: string
}

export type CastRouteParams = {
  movieSlug: string
}
