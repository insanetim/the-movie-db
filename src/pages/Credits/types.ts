import { ErrorMsg } from 'src/interfaces/global.interface'
import { IPersonDetails } from 'src/interfaces/person.interface'

export enum FilterOptions {
  All = 'all',
  Cast = 'cast',
  Crew = 'crew',
}

export type CreditsHookReturn = {
  dataSource: ICredit[]
  error: ErrorMsg
  handleChangeFilter: (filter: FilterOptions) => void
  isLoading: boolean
  person?: IPersonDetails
  personSlug: string
}

export type CreditsRouteParams = {
  personSlug: string
}

export interface ICredit {
  key: string
  movieSlug: string
  posterPath: null | string
  releaseDate: string
  releaseDateTitle: string
  role: string
  title: string
}
