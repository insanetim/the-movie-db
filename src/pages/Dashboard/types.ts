import { Query } from 'src/interfaces/global.interface'

export type DashboardHook = {
  query: Query
}
