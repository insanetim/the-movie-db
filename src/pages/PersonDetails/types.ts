import { ErrorMsg } from 'src/interfaces/global.interface'
import { IPersonDetails } from 'src/interfaces/person.interface'

export type PersonDetailsHookReturn = {
  error: ErrorMsg
  loading: boolean
  person?: IPersonDetails
}

export type PersonRouteParams = {
  personSlug: string
}
