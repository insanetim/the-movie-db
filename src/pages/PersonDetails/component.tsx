import { Button, Col, Divider, Row, Typography } from 'antd'
import { isNil } from 'ramda'
import { Helmet } from 'react-helmet'
import NoImage from 'src/assets/images/no-image.svg'
import KnownFor from 'src/components/PersonDetails/KnownFor'
import PersonalInfo from 'src/components/PersonDetails/PersonalInfo'
import SocialLinks from 'src/components/PersonDetails/SocialLinks'
import Empty from 'src/components/UI/Empty'
import Error from 'src/components/UI/Error'
import Loading from 'src/components/UI/Loading'
import PageTitle from 'src/components/UI/PageTitle'
import isPresent from 'src/utils/helpers/isPresent'
import metaTitle from 'src/utils/helpers/metaTitle'

import useContainer from './hook'

const PersonDetails = () => {
  const { error, handleGoToCredits, loading, person } = useContainer()

  if (loading) {
    return (
      <div className='container top-margin'>
        <Loading />
      </div>
    )
  }

  if (error) {
    return (
      <div className='container top-margin'>
        <Error error={error} />
      </div>
    )
  }

  if (isNil(person)) {
    return <Empty description='Person not found' />
  }

  let poster: JSX.Element
  if (isPresent(person.profile_path)) {
    poster = (
      <div className='person-poster'>
        <img
          alt={person.name}
          src={`https://image.tmdb.org/t/p/w500${person.profile_path}`}
        />
      </div>
    )
  } else {
    poster = (
      <div className='person-poster--no-image'>
        <img
          alt={person.name}
          src={NoImage}
        />
      </div>
    )
  }

  return (
    <>
      <Helmet title={metaTitle(person.name)} />
      <div className='container top-margin'>
        <PageTitle>
          <Typography.Title
            level={1}
            style={{ marginBottom: 0 }}
          >
            {person.name}
          </Typography.Title>
        </PageTitle>
        <Row gutter={24}>
          <Col
            md={8}
            xs={24}
          >
            {poster}
            <SocialLinks externalIds={person.external_ids} />
          </Col>
          <Col
            md={16}
            xs={24}
          >
            <PersonalInfo
              birthday={person.birthday}
              gender={person.gender}
              placeOfBirth={person.place_of_birth}
            />
            {isPresent(person.biography) && (
              <>
                <Divider />
                <Typography.Title level={3}>Biography</Typography.Title>
                <Typography.Paragraph>{person.biography}</Typography.Paragraph>
              </>
            )}
          </Col>
          {(isPresent(person.movie_credits.cast) ||
            isPresent(person.movie_credits.crew)) && (
            <>
              <KnownFor
                credits={person.movie_credits}
                knownForDepartment={person.known_for_department}
              />
              <Col
                className='top-margin text-center'
                span={24}
              >
                <Button
                  ghost
                  onClick={handleGoToCredits}
                  type='primary'
                >
                  Show all credits
                </Button>
              </Col>
            </>
          )}
        </Row>
      </div>
    </>
  )
}

export default PersonDetails
