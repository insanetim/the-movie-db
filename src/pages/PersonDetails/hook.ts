import { useNavigate, useParams } from 'react-router-dom'
import { useGetPersonDetailsQuery } from 'src/store/features/person'
import errorMessage from 'src/utils/helpers/errorMessage'
import getIdFromSlug from 'src/utils/helpers/getIdFromSlug'

import { PersonDetailsHookReturn, PersonDetailsRouteParams } from './types'

const useContainer = (): PersonDetailsHookReturn => {
  const { personSlug } = useParams<
    keyof PersonDetailsRouteParams
  >() as PersonDetailsRouteParams
  const navigate = useNavigate()
  const personId = getIdFromSlug(personSlug)

  const { data: person, error, isLoading } = useGetPersonDetailsQuery(personId)

  const handleGoToCredits = () => {
    navigate('credits')
  }

  return { error: errorMessage(error), handleGoToCredits, isLoading, person }
}

export default useContainer
