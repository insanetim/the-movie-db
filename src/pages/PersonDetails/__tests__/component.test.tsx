import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { mockPersonDetails } from 'src/__mocks__/mockPerson'
import { renderWithWrapper } from 'src/utils/testHelpers/renderWithWrapper'

import PersonDetails from '../component'
import { PersonDetailsHookReturn } from '../types'

const mockedHook: PersonDetailsHookReturn = {
  error: null,
  handleGoToCredits: jest.fn(),
  isLoading: false,
  person: mockPersonDetails,
}
jest.mock('../hook', () => jest.fn(() => mockedHook))

describe('PersonDetails component', () => {
  const user = userEvent.setup()

  it('should match snapshot', () => {
    const { asFragment } = renderWithWrapper(<PersonDetails />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should call "handleGoToCredits" when button clicked', async () => {
    renderWithWrapper(<PersonDetails />)

    const button = screen.getByText('Show all credits')
    await user.click(button)

    expect(mockedHook.handleGoToCredits).toHaveBeenCalled()
  })

  it('should match snapshot with other data', () => {
    mockedHook.person!.profile_path = undefined
    mockedHook.person!.movie_credits.cast = []

    const { asFragment } = renderWithWrapper(<PersonDetails />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with empty person', () => {
    mockedHook.person = undefined

    const { asFragment } = renderWithWrapper(<PersonDetails />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with loading', () => {
    mockedHook.isLoading = true

    const { asFragment } = renderWithWrapper(<PersonDetails />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with error', () => {
    mockedHook.isLoading = false
    mockedHook.error = 'Something went wrong!'

    const { asFragment } = renderWithWrapper(<PersonDetails />)

    expect(asFragment()).toMatchSnapshot()
  })
})
