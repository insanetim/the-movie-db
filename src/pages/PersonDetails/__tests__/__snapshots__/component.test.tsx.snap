// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonDetails component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container top-margin"
  >
    <div
      class="ant-row css-dev-only-do-not-override-vryruh"
    >
      <div
        class="ant-col ant-col-24 css-dev-only-do-not-override-vryruh"
      >
        <div
          class="ant-flex css-dev-only-do-not-override-vryruh ant-flex-align-center ant-flex-gap-middle"
          style="margin-bottom: 24px;"
        >
          <h1
            class="ant-typography css-dev-only-do-not-override-vryruh"
            style="margin-bottom: 0px;"
          >
            test-person
          </h1>
        </div>
      </div>
    </div>
    <div
      class="ant-row css-dev-only-do-not-override-vryruh"
      style="margin-left: -12px; margin-right: -12px;"
    >
      <div
        class="ant-col ant-col-xs-24 ant-col-md-8 css-dev-only-do-not-override-vryruh"
        style="padding-left: 12px; padding-right: 12px;"
      >
        <div
          class="person-poster"
        >
          <img
            alt="test-person"
            src="https://image.tmdb.org/t/p/w500/image"
          />
        </div>
        <div
          class="social-link__list ant-flex css-dev-only-do-not-override-vryruh"
          style="gap: 16px;"
        >
          <a
            class="social-link"
            href="https://facebook.com/facebook_user"
            rel="noreferrer"
            target="_blank"
          >
            <span
              aria-label="facebook"
              class="anticon anticon-facebook"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="facebook"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-92.4 233.5h-63.9c-50.1 0-59.8 23.8-59.8 58.8v77.1h119.6l-15.6 120.7h-104V912H539.2V602.2H434.9V481.4h104.3v-89c0-103.3 63.1-159.6 155.3-159.6 44.2 0 82.1 3.3 93.2 4.8v107.9z"
                />
              </svg>
            </span>
          </a>
          <a
            class="social-link"
            href="https://instagram.com/instagram_user"
            rel="noreferrer"
            target="_blank"
          >
            <span
              aria-label="instagram"
              class="anticon anticon-instagram"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="instagram"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 306.9c-113.5 0-205.1 91.6-205.1 205.1S398.5 717.1 512 717.1 717.1 625.5 717.1 512 625.5 306.9 512 306.9zm0 338.4c-73.4 0-133.3-59.9-133.3-133.3S438.6 378.7 512 378.7 645.3 438.6 645.3 512 585.4 645.3 512 645.3zm213.5-394.6c-26.5 0-47.9 21.4-47.9 47.9s21.4 47.9 47.9 47.9 47.9-21.3 47.9-47.9a47.84 47.84 0 00-47.9-47.9zM911.8 512c0-55.2.5-109.9-2.6-165-3.1-64-17.7-120.8-64.5-167.6-46.9-46.9-103.6-61.4-167.6-64.5-55.2-3.1-109.9-2.6-165-2.6-55.2 0-109.9-.5-165 2.6-64 3.1-120.8 17.7-167.6 64.5C132.6 226.3 118.1 283 115 347c-3.1 55.2-2.6 109.9-2.6 165s-.5 109.9 2.6 165c3.1 64 17.7 120.8 64.5 167.6 46.9 46.9 103.6 61.4 167.6 64.5 55.2 3.1 109.9 2.6 165 2.6 55.2 0 109.9.5 165-2.6 64-3.1 120.8-17.7 167.6-64.5 46.9-46.9 61.4-103.6 64.5-167.6 3.2-55.1 2.6-109.8 2.6-165zm-88 235.8c-7.3 18.2-16.1 31.8-30.2 45.8-14.1 14.1-27.6 22.9-45.8 30.2C695.2 844.7 570.3 840 512 840c-58.3 0-183.3 4.7-235.9-16.1-18.2-7.3-31.8-16.1-45.8-30.2-14.1-14.1-22.9-27.6-30.2-45.8C179.3 695.2 184 570.3 184 512c0-58.3-4.7-183.3 16.1-235.9 7.3-18.2 16.1-31.8 30.2-45.8s27.6-22.9 45.8-30.2C328.7 179.3 453.7 184 512 184s183.3-4.7 235.9 16.1c18.2 7.3 31.8 16.1 45.8 30.2 14.1 14.1 22.9 27.6 30.2 45.8C844.7 328.7 840 453.7 840 512c0 58.3 4.7 183.2-16.2 235.8z"
                />
              </svg>
            </span>
          </a>
          <a
            class="social-link"
            href="https://tiktok.com/@tiktok_user"
            rel="noreferrer"
            target="_blank"
          >
            <span
              aria-label="tik-tok"
              class="anticon anticon-tik-tok"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="tik-tok"
                fill="currentColor"
                fill-rule="evenodd"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M912 224.96C912 162.57 861.42 112 799.04 112H224.96C162.57 112 112 162.57 112 224.96v574.08C112 861.43 162.58 912 224.96 912h574.08C861.42 912 912 861.43 912 799.04zM774.76 460.92c-51.62.57-99.71-15.03-141.94-43.93v202.87a192.3 192.3 0 01-149 187.85c-119.06 27.17-219.86-58.95-232.57-161.83-13.3-102.89 52.32-193.06 152.89-213.29 19.65-4.04 49.2-4.04 64.46-.57v108.66c-4.7-1.15-9.09-2.31-13.71-2.89-39.3-6.94-77.37 12.72-92.98 48.55-15.6 35.84-5.16 77.45 26.63 101.73 26.59 20.8 56.09 23.7 86.14 9.82 30.06-13.29 46.21-37.56 49.68-70.5.58-4.63.54-9.84.54-15.04V222.21c0-10.99.09-10.5 11.07-10.5h86.12c6.36 0 8.67.9 9.25 8.43 4.62 67.04 55.53 124.14 120.84 132.81 6.94 1.16 14.37 1.62 22.58 2.2z"
                />
              </svg>
            </span>
          </a>
          <a
            class="social-link"
            href="https://twitter.com/twitter_user"
            rel="noreferrer"
            target="_blank"
          >
            <span
              aria-label="twitter"
              class="anticon anticon-twitter"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="twitter"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M928 254.3c-30.6 13.2-63.9 22.7-98.2 26.4a170.1 170.1 0 0075-94 336.64 336.64 0 01-108.2 41.2A170.1 170.1 0 00672 174c-94.5 0-170.5 76.6-170.5 170.6 0 13.2 1.6 26.4 4.2 39.1-141.5-7.4-267.7-75-351.6-178.5a169.32 169.32 0 00-23.2 86.1c0 59.2 30.1 111.4 76 142.1a172 172 0 01-77.1-21.7v2.1c0 82.9 58.6 151.6 136.7 167.4a180.6 180.6 0 01-44.9 5.8c-11.1 0-21.6-1.1-32.2-2.6C211 652 273.9 701.1 348.8 702.7c-58.6 45.9-132 72.9-211.7 72.9-14.3 0-27.5-.5-41.2-2.1C171.5 822 261.2 850 357.8 850 671.4 850 843 590.2 843 364.7c0-7.4 0-14.8-.5-22.2 33.2-24.3 62.3-54.4 85.5-88.2z"
                />
              </svg>
            </span>
          </a>
        </div>
      </div>
      <div
        class="ant-col ant-col-xs-24 ant-col-md-16 css-dev-only-do-not-override-vryruh"
        style="padding-left: 12px; padding-right: 12px;"
      >
        <h3
          class="ant-typography css-dev-only-do-not-override-vryruh"
        >
          Personal Info
        </h3>
        <div
          class="ant-typography css-dev-only-do-not-override-vryruh"
        >
          <b>
            Gender: 
          </b>
          <span>
            Not set / not specified
          </span>
        </div>
        <div
          class="ant-typography css-dev-only-do-not-override-vryruh"
        >
          <b>
            Birthday: 
          </b>
          <span>
            January 1, 1999 (25 years old)
          </span>
        </div>
        <div
          class="ant-typography css-dev-only-do-not-override-vryruh"
        >
          <b>
            Place of Birth: 
          </b>
          <span>
            New York, USA
          </span>
        </div>
        <div
          class="ant-divider css-dev-only-do-not-override-vryruh ant-divider-horizontal"
          role="separator"
        />
        <h3
          class="ant-typography css-dev-only-do-not-override-vryruh"
        >
          Biography
        </h3>
        <div
          class="ant-typography css-dev-only-do-not-override-vryruh"
        >
          test-biography
        </div>
      </div>
      <div
        class="ant-col ant-col-24 top-margin css-dev-only-do-not-override-vryruh"
        style="padding-left: 12px; padding-right: 12px;"
      >
        <h2
          class="ant-typography css-dev-only-do-not-override-vryruh"
        >
          Known For
        </h2>
      </div>
      <div
        class="ant-col ant-col-24 css-dev-only-do-not-override-vryruh"
        style="padding-left: 12px; padding-right: 12px;"
      >
        <div
          class="ant-row css-dev-only-do-not-override-vryruh"
          style="margin-left: -12px; margin-right: -12px; row-gap: 16px;"
        >
          <div
            class="ant-col ant-col-24 ant-col-sm-12 ant-col-md-8 ant-col-lg-8 ant-col-xl-6 css-dev-only-do-not-override-vryruh"
            style="padding-left: 12px; padding-right: 12px;"
          >
            <div
              class="ant-card ant-card-bordered ant-card-hoverable css-dev-only-do-not-override-vryruh"
              data-testid="movieItemCard"
            >
              <div
                class="ant-card-cover"
              >
                <img
                  alt="The Matrix"
                  height="750"
                  loading="lazy"
                  src="https://image.tmdb.org/t/p/w500/image"
                  width="500"
                />
              </div>
              <div
                class="ant-card-body"
              >
                <div
                  class="ant-card-meta"
                >
                  <div
                    class="ant-card-meta-detail"
                  >
                    <div
                      class="ant-card-meta-title"
                    >
                      The Matrix
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col ant-col-24 ant-col-sm-12 ant-col-md-8 ant-col-lg-8 ant-col-xl-6 css-dev-only-do-not-override-vryruh"
            style="padding-left: 12px; padding-right: 12px;"
          >
            <div
              class="ant-card ant-card-bordered ant-card-hoverable css-dev-only-do-not-override-vryruh"
              data-testid="movieItemCard"
            >
              <div
                class="ant-card-cover"
              >
                <img
                  alt="The Matrix Reloaded"
                  height="750"
                  loading="lazy"
                  src="https://image.tmdb.org/t/p/w500/image"
                  width="500"
                />
              </div>
              <div
                class="ant-card-body"
              >
                <div
                  class="ant-card-meta"
                >
                  <div
                    class="ant-card-meta-detail"
                  >
                    <div
                      class="ant-card-meta-title"
                    >
                      The Matrix Reloaded
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-24 top-margin text-center css-dev-only-do-not-override-vryruh"
        style="padding-left: 12px; padding-right: 12px;"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-vryruh ant-btn-primary ant-btn-color-primary ant-btn-variant-solid ant-btn-background-ghost"
          type="button"
        >
          <span>
            Show all credits
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PersonDetails component should match snapshot with empty person 1`] = `
<DocumentFragment>
  <div
    class="css-dev-only-do-not-override-vryruh ant-empty ant-empty-normal"
  >
    <div
      class="ant-empty-image"
    >
      <svg
        height="41"
        viewBox="0 0 64 41"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Simple Empty
        </title>
        <g
          fill="none"
          fill-rule="evenodd"
          transform="translate(0 1)"
        >
          <ellipse
            cx="32"
            cy="33"
            fill="#f5f5f5"
            rx="32"
            ry="7"
          />
          <g
            fill-rule="nonzero"
            stroke="#d9d9d9"
          >
            <path
              d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
            />
            <path
              d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
              fill="#fafafa"
            />
          </g>
        </g>
      </svg>
    </div>
    <div
      class="ant-empty-description"
    >
      Person not found
    </div>
  </div>
</DocumentFragment>
`;

exports[`PersonDetails component should match snapshot with error 1`] = `
<DocumentFragment>
  <div
    class="container top-margin"
  >
    <div
      class="ant-result ant-result-error css-dev-only-do-not-override-vryruh"
    >
      <div
        class="ant-result-icon"
      >
        <span
          aria-label="close-circle"
          class="anticon anticon-close-circle"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="close-circle"
            fill="currentColor"
            fill-rule="evenodd"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
            />
          </svg>
        </span>
      </div>
      <div
        class="ant-result-title"
      >
        Something went wrong!
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PersonDetails component should match snapshot with loading 1`] = `
<DocumentFragment>
  <div
    class="container top-margin"
  >
    <div
      class="ant-row ant-row-center top-margin css-dev-only-do-not-override-vryruh"
    >
      <div
        class="ant-col css-dev-only-do-not-override-vryruh"
      >
        <div
          aria-busy="true"
          aria-live="polite"
          class="ant-spin ant-spin-spinning css-dev-only-do-not-override-vryruh"
        >
          <span
            class="ant-spin-dot-holder"
          >
            <span
              class="ant-spin-dot ant-spin-dot-spin"
            >
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PersonDetails component should match snapshot with other data 1`] = `
<DocumentFragment>
  <div
    class="container top-margin"
  >
    <div
      class="ant-row css-dev-only-do-not-override-vryruh"
    >
      <div
        class="ant-col ant-col-24 css-dev-only-do-not-override-vryruh"
      >
        <div
          class="ant-flex css-dev-only-do-not-override-vryruh ant-flex-align-center ant-flex-gap-middle"
          style="margin-bottom: 24px;"
        >
          <h1
            class="ant-typography css-dev-only-do-not-override-vryruh"
            style="margin-bottom: 0px;"
          >
            test-person
          </h1>
        </div>
      </div>
    </div>
    <div
      class="ant-row css-dev-only-do-not-override-vryruh"
      style="margin-left: -12px; margin-right: -12px;"
    >
      <div
        class="ant-col ant-col-xs-24 ant-col-md-8 css-dev-only-do-not-override-vryruh"
        style="padding-left: 12px; padding-right: 12px;"
      >
        <div
          class="person-poster--no-image"
        >
          <img
            alt="test-person"
            src="test-file-stub"
          />
        </div>
        <div
          class="social-link__list ant-flex css-dev-only-do-not-override-vryruh"
          style="gap: 16px;"
        >
          <a
            class="social-link"
            href="https://facebook.com/facebook_user"
            rel="noreferrer"
            target="_blank"
          >
            <span
              aria-label="facebook"
              class="anticon anticon-facebook"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="facebook"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-92.4 233.5h-63.9c-50.1 0-59.8 23.8-59.8 58.8v77.1h119.6l-15.6 120.7h-104V912H539.2V602.2H434.9V481.4h104.3v-89c0-103.3 63.1-159.6 155.3-159.6 44.2 0 82.1 3.3 93.2 4.8v107.9z"
                />
              </svg>
            </span>
          </a>
          <a
            class="social-link"
            href="https://instagram.com/instagram_user"
            rel="noreferrer"
            target="_blank"
          >
            <span
              aria-label="instagram"
              class="anticon anticon-instagram"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="instagram"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 306.9c-113.5 0-205.1 91.6-205.1 205.1S398.5 717.1 512 717.1 717.1 625.5 717.1 512 625.5 306.9 512 306.9zm0 338.4c-73.4 0-133.3-59.9-133.3-133.3S438.6 378.7 512 378.7 645.3 438.6 645.3 512 585.4 645.3 512 645.3zm213.5-394.6c-26.5 0-47.9 21.4-47.9 47.9s21.4 47.9 47.9 47.9 47.9-21.3 47.9-47.9a47.84 47.84 0 00-47.9-47.9zM911.8 512c0-55.2.5-109.9-2.6-165-3.1-64-17.7-120.8-64.5-167.6-46.9-46.9-103.6-61.4-167.6-64.5-55.2-3.1-109.9-2.6-165-2.6-55.2 0-109.9-.5-165 2.6-64 3.1-120.8 17.7-167.6 64.5C132.6 226.3 118.1 283 115 347c-3.1 55.2-2.6 109.9-2.6 165s-.5 109.9 2.6 165c3.1 64 17.7 120.8 64.5 167.6 46.9 46.9 103.6 61.4 167.6 64.5 55.2 3.1 109.9 2.6 165 2.6 55.2 0 109.9.5 165-2.6 64-3.1 120.8-17.7 167.6-64.5 46.9-46.9 61.4-103.6 64.5-167.6 3.2-55.1 2.6-109.8 2.6-165zm-88 235.8c-7.3 18.2-16.1 31.8-30.2 45.8-14.1 14.1-27.6 22.9-45.8 30.2C695.2 844.7 570.3 840 512 840c-58.3 0-183.3 4.7-235.9-16.1-18.2-7.3-31.8-16.1-45.8-30.2-14.1-14.1-22.9-27.6-30.2-45.8C179.3 695.2 184 570.3 184 512c0-58.3-4.7-183.3 16.1-235.9 7.3-18.2 16.1-31.8 30.2-45.8s27.6-22.9 45.8-30.2C328.7 179.3 453.7 184 512 184s183.3-4.7 235.9 16.1c18.2 7.3 31.8 16.1 45.8 30.2 14.1 14.1 22.9 27.6 30.2 45.8C844.7 328.7 840 453.7 840 512c0 58.3 4.7 183.2-16.2 235.8z"
                />
              </svg>
            </span>
          </a>
          <a
            class="social-link"
            href="https://tiktok.com/@tiktok_user"
            rel="noreferrer"
            target="_blank"
          >
            <span
              aria-label="tik-tok"
              class="anticon anticon-tik-tok"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="tik-tok"
                fill="currentColor"
                fill-rule="evenodd"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M912 224.96C912 162.57 861.42 112 799.04 112H224.96C162.57 112 112 162.57 112 224.96v574.08C112 861.43 162.58 912 224.96 912h574.08C861.42 912 912 861.43 912 799.04zM774.76 460.92c-51.62.57-99.71-15.03-141.94-43.93v202.87a192.3 192.3 0 01-149 187.85c-119.06 27.17-219.86-58.95-232.57-161.83-13.3-102.89 52.32-193.06 152.89-213.29 19.65-4.04 49.2-4.04 64.46-.57v108.66c-4.7-1.15-9.09-2.31-13.71-2.89-39.3-6.94-77.37 12.72-92.98 48.55-15.6 35.84-5.16 77.45 26.63 101.73 26.59 20.8 56.09 23.7 86.14 9.82 30.06-13.29 46.21-37.56 49.68-70.5.58-4.63.54-9.84.54-15.04V222.21c0-10.99.09-10.5 11.07-10.5h86.12c6.36 0 8.67.9 9.25 8.43 4.62 67.04 55.53 124.14 120.84 132.81 6.94 1.16 14.37 1.62 22.58 2.2z"
                />
              </svg>
            </span>
          </a>
          <a
            class="social-link"
            href="https://twitter.com/twitter_user"
            rel="noreferrer"
            target="_blank"
          >
            <span
              aria-label="twitter"
              class="anticon anticon-twitter"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="twitter"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M928 254.3c-30.6 13.2-63.9 22.7-98.2 26.4a170.1 170.1 0 0075-94 336.64 336.64 0 01-108.2 41.2A170.1 170.1 0 00672 174c-94.5 0-170.5 76.6-170.5 170.6 0 13.2 1.6 26.4 4.2 39.1-141.5-7.4-267.7-75-351.6-178.5a169.32 169.32 0 00-23.2 86.1c0 59.2 30.1 111.4 76 142.1a172 172 0 01-77.1-21.7v2.1c0 82.9 58.6 151.6 136.7 167.4a180.6 180.6 0 01-44.9 5.8c-11.1 0-21.6-1.1-32.2-2.6C211 652 273.9 701.1 348.8 702.7c-58.6 45.9-132 72.9-211.7 72.9-14.3 0-27.5-.5-41.2-2.1C171.5 822 261.2 850 357.8 850 671.4 850 843 590.2 843 364.7c0-7.4 0-14.8-.5-22.2 33.2-24.3 62.3-54.4 85.5-88.2z"
                />
              </svg>
            </span>
          </a>
        </div>
      </div>
      <div
        class="ant-col ant-col-xs-24 ant-col-md-16 css-dev-only-do-not-override-vryruh"
        style="padding-left: 12px; padding-right: 12px;"
      >
        <h3
          class="ant-typography css-dev-only-do-not-override-vryruh"
        >
          Personal Info
        </h3>
        <div
          class="ant-typography css-dev-only-do-not-override-vryruh"
        >
          <b>
            Gender: 
          </b>
          <span>
            Not set / not specified
          </span>
        </div>
        <div
          class="ant-typography css-dev-only-do-not-override-vryruh"
        >
          <b>
            Birthday: 
          </b>
          <span>
            January 1, 1999 (25 years old)
          </span>
        </div>
        <div
          class="ant-typography css-dev-only-do-not-override-vryruh"
        >
          <b>
            Place of Birth: 
          </b>
          <span>
            New York, USA
          </span>
        </div>
        <div
          class="ant-divider css-dev-only-do-not-override-vryruh ant-divider-horizontal"
          role="separator"
        />
        <h3
          class="ant-typography css-dev-only-do-not-override-vryruh"
        >
          Biography
        </h3>
        <div
          class="ant-typography css-dev-only-do-not-override-vryruh"
        >
          test-biography
        </div>
      </div>
      <div
        class="ant-col ant-col-24 top-margin css-dev-only-do-not-override-vryruh"
        style="padding-left: 12px; padding-right: 12px;"
      >
        <h2
          class="ant-typography css-dev-only-do-not-override-vryruh"
        >
          Known For
        </h2>
      </div>
      <div
        class="ant-col ant-col-24 css-dev-only-do-not-override-vryruh"
        style="padding-left: 12px; padding-right: 12px;"
      >
        <div
          class="ant-row css-dev-only-do-not-override-vryruh"
          style="margin-left: -12px; margin-right: -12px; row-gap: 16px;"
        />
      </div>
      <div
        class="ant-col ant-col-24 top-margin text-center css-dev-only-do-not-override-vryruh"
        style="padding-left: 12px; padding-right: 12px;"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-vryruh ant-btn-primary ant-btn-color-primary ant-btn-variant-solid ant-btn-background-ghost"
          type="button"
        >
          <span>
            Show all credits
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
